x-pack/plugins/observability_solution/infra/public/components/asset_details/hooks/use_update_custom_dashboards.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import { fold } from 'fp-ts/lib/Either';
+import { identity } from 'fp-ts/lib/function';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import type { InfraCustomDashboard } from '../../../../common/custom_dashboards';
+import { InfraCustomDashboardRT } from '../../../../common/http_api/custom_dashboards_api';
+import { throwErrors, createPlainError } from '../../../../common/runtime_types';
+
+export function useUpdateCustomDashboard() {
+  const { services } = useKibanaContextForPlugin();
+
+  const decodeResponse = (response: any) => {
+    return pipe(
+      InfraCustomDashboardRT.decode(response),
+      fold(throwErrors(createPlainError), identity)
+    );
+  };
+
+  const [updateCustomDashboardRequest, updateCustomDashboard] = useTrackedPromise(
+    {
+      cancelPreviousOn: 'resolution',
+      createPromise: async ({ assetType, dashboardIdList }: InfraCustomDashboard) => {
+        const rawResponse = await services.http.fetch('/api/infra/custom-dashboards', {
+          method: 'POST',
```

Comment:
So the API supports `GET` and `POST` (https://github.com/elastic/kibana/pull/176612) The idea here is that we will update the list by changing the array (adding/removing the dashboard list items) so to support removing items and creating the initial saved object we need the endpoint to support `POST` - so the logic behind it [creates or updates the saved object](https://github.com/mykolaharmash/kibana/blob/0dd6e8201220615fb5306c9e1b1b46b61bd34a1d/x-pack/plugins/infra/server/routes/custom_dashboards/save_custom_dashboard.ts#L43-L65). If we use it to only link dashboards to a specific asset type assuming it exists it can be `PUT` (We can also have both `PUT` for linking new dashboards after we have the saved object in place and `POST` to use when unlinking a dashboard or adding the first dashboard per asset type but that's not how the API was designed - also it overcomplicates the logic there so I would keep it as `POST`)